common/utilities/tests/test_timing.py

```diff
@@ -0,0 +1,207 @@
+
+from unittest import TestCase
+from unittest.mock import patch, Mock
+
+from tornado.testing import AsyncHTTPTestCase
+from tornado.web import Application, RequestHandler
+
+from common.utilities import timing
+from common.utilities.test_utilities import async_test
+
+
+class TestTimeUtilities(TestCase):
+
+    @patch('time.perf_counter')
+    def test_stopwatch(self, time_mock):
+        stopwatch = timing.Stopwatch()
+        time_mock.return_value = 0.0
+        stopwatch.start_timer()
+        time_mock.return_value = 5.0
+        result = stopwatch.stop_timer()
+
+        self.assertEqual(result, 5)
+
+    @patch('logging.info')
+    def test_invoke_with_time_rounding(self, log_mock):
+        timing._log_time(5.1236, "yes")
+        log_mock.assert_called_with('Method name=yes took duration=5.124')
+
+    @patch('common.utilities.timing._log_time')
+    @patch('common.utilities.timing.Stopwatch.stop_timer')
+    @async_test
+    async def test_invoke_with_time(self, time_mock, log_mock):
+        time_mock.return_value = 5
+        with self.subTest("Sync version"):
+            res = self.default_method()
+            log_mock.assert_called_with(5, 'default_method')
+            self.assertEqual("slept", res)
+
+        with self.subTest("Async version"):
+            res = await self.default_method_async()
+            log_mock.assert_called_with(5, 'default_method_async')
+            self.assertEqual(5, res)
+
+    @patch('common.utilities.timing.Stopwatch.stop_timer')
+    @patch('common.utilities.timing._log_time')
+    @async_test
+    async def test_exception_thrown_whilst_timing(self, log_mock, time_mock):
+        time_mock.return_value = 5
+        with self.subTest("Sync"):
+            try:
+                self.throw_error_method()
+            except ValueError:
+                pass
```

Comment:
It might make more sense to do `with self.assertRaises(ValueError): self.throw_error_method()`, as then the test fails if the exception wasn't thrown as expected.